backEnd/src/main/java/phylogenetictree/PhylogeneticNode.java

```diff
@@ -24,21 +30,41 @@ public PhylogeneticNode() {
         nameLabel = "";
         distance = 0.;
         children = new ArrayList<>();
+        genomes= new ArrayList<>();
     }
 
     /**
-     * Initialize this node from tree node.
+     * Initialize this node from tree node, recursively.
      * @param node The node.
+     * @param parent The parentNode of this node, null if root.
      * @param distance The distance.
+     * @param id The id of this node, root being 0. Incrementing breath first.
      */
-    public PhylogeneticNode(TreeNode node, double distance) {
+    public PhylogeneticNode(TreeNode node, PhylogeneticNode parent, double distance, int id) {
```

Comment:
This could probably be resolved in a cleaner way. Now this field also gets sent as data to the client during JSON translation even though its not needed.